File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/adverts/DeleteAdvertHandler.java
package by.fpmibsu.findafriend.controller.commands.adverts;

import by.fpmibsu.findafriend.application.Application;
import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.advert.AdvertDao;

public class DeleteAdvertHandler extends RequestHandler<Boolean, DeleteAdvertCommand> {
    private final AdvertDao advertDao;

    public DeleteAdvertHandler(AdvertDao userDao) {
        this.advertDao = userDao;
    }

    @Override
    public Boolean handle(DeleteAdvertCommand request) {
        return advertDao.deleteAdvertById(request.getAdvertId());
    }
}

File: BackendNET/Backend/Commands/Adverts/DeleteAdvert/DeleteAdvertHandler.cs
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Adverts.DeleteAdvert;

public class DeleteAdvertHandler : IRequestHandler<DeleteAdvertCommand, bool>
{
    private readonly IAdvertRepository _repository;

    public DeleteAdvertHandler(IAdvertRepository repository)
    {
        _repository = repository;
    }

    public async Task<bool> Handle(DeleteAdvertCommand request, CancellationToken cancellationToken)
    {
        return await _repository.DeleteAsync(request.AdvertId, cancellationToken) == 1;
    }
}